### DISCUSSION TITLE
Code folding?

### DISCUSSION THREAD

**Discussion by pusolito** [code-hike/codehike#310] [Q&A]
Is code folding supported or planned?

**Comment by pomber** [comment]
It's something I want to add, but it's hard for technical reasons. So, it's not planned for the near future.

**Reply by pusolito** [reply]
Can you elaborate? Is this challenging to solve at the language level? I ask because there's a potential to solve part of this with new annotations that let the author specify where to introduce a fold. This could work like `focus` in that it takes a range.

**Reply by pomber** [reply]
It's challenging because a lot of existing code assumes that the height is proportional to the number of lines, and folding break that assumption. It's doable, but it's hard because of the existing code.

**Reply by eric-horodyski** [reply]
I definitely second this, and wouldn't be opposed to only being able to use it unless the `rows` property is defined on the parent component, something like the mockup below. Since `<CH.Scrollycoding>` is min/max height based, maybe that's a scenario where you wouldn't need to supply a defined height in rows.

I would find this extremely beneficial. A really good example of why is the following link: https://eric-horodyski.github.io/ionic-tutorials-poc/#get-the-tokens. In this example, I'm having the reader add some utility methods to a React Context/Provider combo, and there's only enough room to highlight the Context's "interface", leaving it up to the reader to assume they need to scroll down for the method implementations.

![untitled](https://user-images.githubusercontent.com/62819287/227583406-3f468986-04b2-42b2-a74a-2d4ec4da5e8d.png)
